packages/cli/src/commands/config/help.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command } from 'commander';
+import { JsonObject } from '@backstage/config';
+import { loadCliConfig } from '../../lib/config';
+
+export default async (options: string[], cmd: Command) => {
+  const { schema } = await loadCliConfig({
+    args: [],
+    fromPackage: cmd.package,
+    mockEnv: true,
+  });
+
+  const data = schema.serialize();
+  const schemas = (data.schemas as JsonObject[]) || [];
+  const opts_specs =
+    options.length > 0
+      ? options.map((option: string): [string, JsonObject[]] => [
+          option,
+          Array.from(lookupOption(schemas, option)),
+        ])
+      : schemas.map((plugin_schema): [string, JsonObject[]] => [
+          '<root>',
+          [
+            {
+              _source: plugin_schema?.path,
+              _required: true,
+              ...(plugin_schema?.value as JsonObject),
+            },
+          ],
+        ]);
+
+  for (const [option, specs] of opts_specs) {
+    if (specs.length === 0) {
+      process.stdout.write(`Unknown configuration option: ${option}\n---\n`);
+    } else {
+      for (const spec of specs) {
+        printConfigHelp(option, spec);
+      }
+    }
+  }
+};
+
+function* lookupOption(schemas: JsonObject[], option: string) {
+  const key_path = option ? option.split('.') : [];
+
+  for (const schema of schemas) {
+    const spec = resolveKey((schema?.value as JsonObject) || {}, key_path);
+    if (spec !== undefined) {
+      spec._source = schema?.path;
+      yield spec;
+    }
+  }
+}
+
+function resolveKey(obj: JsonObject, path: string[]): JsonObject | undefined {
```

Comment:
This kind of traversal is a bit tricky when considering keywords like `anyOf`, `allOf`, `not`, etc. If you for example look at the schema for `backend.https.certificate` you'll find that it's behind an `anyOf`, as `backend.https` can also be set to `true`. You'll find the way that's defined here: https://github.com/backstage/backstage/blob/c600680599011f2c17514886a34ed25a37bff27d/packages/backend-common/config.d.ts#L41

There's quite a lot of usage of those keywords, especially `anyOf`, so I do think it would be best if we supported it to avoid confusion. Best case I think would be if there's some existing library that is able to drill into a schema object given a path, maybe even `jsonschema` or `ajv` is able to help us there? x)